src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -3192,17 +3230,29 @@ void AwtWindow::_SetOpaque(void* param)
 {
     JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    OpaqueStruct *os = (OpaqueStruct *)param;
+    OpaqueStruct *os = static_cast<OpaqueStruct *>(param);
     jobject self = os->window;
     BOOL isOpaque = (BOOL)os->isOpaque;
 
-    PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
-    AwtWindow *window = (AwtWindow *)pData;
+    AwtWindow *window = nullptr;
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        delete os;
+        return;
+    } else {
+        window = (AwtWindow *) JNI_GET_PDATA(self);
+        if (window == nullptr) {
+            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
+            env->DeleteGlobalRef(self);
+            delete os;
+            return;
+        }
+    }
 
     window->SetTranslucency(window->getOpacity(), isOpaque);
 
-  ret:
+
```

Comment:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.